azure_monitor/azfunc_sample/README.md

```diff
@@ -0,0 +1,86 @@
+---
+page_type: sample
+languages:
+- python
+products:
+- azure
+description: "This sample contains a simple Azure Function to show how you can send correlated application log information as well as information of external dependencies to Azure Monitor (Application Insights)."
+urlFragment: azure-monitor-opencensus-python
+---
+
+# Correlated End-To-End Azure Function (Python) Instrumentation with Opencensus 
+
+## Overview
+
+Azure Function (Python) already supports instrumentaton using Opencensus. Unfortunately, the current out-of-the-box (OOTB) implementation/support has some flaws (see below), which make it difficult to get a correlated end-to-end view of a single function invocation (in Azure Monitor/Application Insights).
+
+It is also important to assign the correct cloud role name and cloud instance name to different log records sent to Application Insights (for additional insights). This is especially relevant for distributed, scalable applications sending logs to a single Application Insights instance.
+
+### Example of a Correlated End-to-End Azure Function Invocation
+
+End-to-end transaction:
+
+![AI Correlated end2end function invocation](./docs/assets/01-ai-correlated-invocation.PNG)
+
+Associated application trace messages:
+
+![AI Application trace messages](./docs/assets/02-ai-correlated-invocation.PNG)
+
+Associated AI Application Map:
+
+![AI Application Map](./docs/assets/03-ai-application-map.PNG)
+
+The images shows the following elements:
+
+- Correlated (per Azure Function invocatio) log information incl
+    - Incoming request information
+    - Appication traces & error messages
+    - External dependencies (IN-PROC & HTTP using Python's requests)
+- Application Map (defined cloud role and cloud instance values)
+
+### Known Limitations in Current OOTB Logging Integration (Azure Function)
+
+- External dependency log records don't have the correct *Cloud Role Name* (field: cloud_RoleName). This results in wrong information displayed in Application Insight's *Application Map*. See <https://github.com/census-instrumentation/opencensus-python/issues/1052>
```

Comment:
This is not a limitation. Users can configure cloud role name via [telemetry processors](https://github.com/census-instrumentation/opencensus-python/issues/1052)